### PR TITLE
clone codec per worker

### PR DESCRIPTION
**Pull request by colinsurprenant**
Per discussion in elastic/logstash/issues/8830 each input worker should have its own codec instance.

### CODE REVIEW

**Comment by jordansissel** [comment]
code looks ok. Any thoughts on adding a test? Maybe something that sends 2 lines in a single UDP packet and asserts that 2 events come out? Also that 2 events (line delimited w/ line codec) per packet, sending 2 packets, results in 4 events?

**Comment by colinsurprenant** [comment]
@jordansissel yeah, good idea, will do.

**Comment by colinsurprenant** [comment]
@jordansissel added new test, LMKWYT.

**Comment by jordansissel** [comment]
Delightful, thank you!

LGTM

File: spec/inputs/udp_spec.rb [hunk 1]
```diff
@@ -46,10 +46,28 @@
         expect(message).to match(/msg \d+/)
       end
     end
-
   end
 
-  it_behaves_like "an interruptible input plugin" do
```
Comment (yaauie) [inline spec/inputs/udp_spec.rb]: Why are we removing these specs?

**Comment by jordansissel** [comment]
Good catch, I didn't notice that removal. A test for shutdown should still be included (https://github.com/elastic/logstash-devutils/blob/master/lib/logstash/devutils/rspec/shared_examples.rb).

**Comment by colinsurprenant** [comment]
@jordansissel @yaauie OMG good catch - dunno why/how I removed that, a mistake.

**Comment by colinsurprenant** [comment]
Added deleted spec, assuming it now LGTM per your comments - thanks @jordansissel  @yaauie

**Comment by colinsurprenant** [comment]
rebased changes, added version bump commit, will merge and publish.

**Comment by jordansissel** [comment]
LGTM

**Pull request merged**